### PR TITLE
More missing AI logic

### PR DESCRIPTION
**Pull request by kittenchilly**
Description
#5268 but targeting upcoming
Added missing ai logic for Population Bomb, Doodle, and Corrosive Gas. Also fixed some things with role play and moves that deal a set amount of hp

Discord contact info
kittenchilly

### CODE REVIEW

File: src/battle_ai_main.c [hunk 1]
```diff
@@ -2176,6 +2179,17 @@ static s32 AI_CheckBadMove(u32 battlerAtk, u32 battlerDef, u32 move, s32 score)
             else if (IsAbilityOfRating(aiData->abilities[battlerAtk], 5))
                 ADJUST_SCORE(-4);
             break;
+        case EFFECT_DOODLE: // Same as Role Play, but also check if the partner's ability should be replaced
+            if (aiData->abilities[battlerAtk] == aiData->abilities[battlerDef]
+              || aiData->abilities[BATTLE_PARTNER(battlerAtk)] == aiData->abilities[battlerDef]
+              || aiData->abilities[battlerDef] == ABILITY_NONE
+              || gAbilitiesInfo[aiData->abilities[battlerAtk]].cantBeSuppressed
+              || gAbilitiesInfo[aiData->abilities[BATTLE_PARTNER(battlerAtk)]].cantBeSuppressed
+              || gAbilitiesInfo[aiData->abilities[battlerDef]].cantBeCopied)
```
Comment (AlexOn1ine) [inline src/battle_ai_main.c:2188]:
```suggestion
            if (aiData->abilities[battlerAtk] == aiData->abilities[battlerDef]
             || aiData->abilities[BATTLE_PARTNER(battlerAtk)] == aiData->abilities[battlerDef]
             || aiData->abilities[battlerDef] == ABILITY_NONE
             || gAbilitiesInfo[aiData->abilities[battlerAtk]].cantBeSuppressed
             || gAbilitiesInfo[aiData->abilities[BATTLE_PARTNER(battlerAtk)]].cantBeSuppressed
             || gAbilitiesInfo[aiData->abilities[battlerDef]].cantBeCopied)
```

Comment (kittenchilly) [reply src/battle_ai_main.c:2188]: Its not like this for any other line in this file

Comment (AlexOn1ine) [reply src/battle_ai_main.c:2188]: Doesn't matter. If you make changes to it we might as well do proper alignment.

Comment (kittenchilly) [reply src/battle_ai_main.c:2188]: :/ OK

**Review by AlexOn1ine — APPROVED** [review]

**Pull request merged**